low_level_platform/api/low_level_platform.h

```diff
@@ -132,6 +136,50 @@ int lf_thread_create(lf_thread_t* thread, void* (*lf_thread)(void*), void* argum
  */
 int lf_thread_join(lf_thread_t thread, void** thread_return);
 
+#define LF_SCHED_MAX_PRIORITY 99
+#define LF_SCHED_MIN_PRIORITY 0
+/**
+ * @brief The thread scheduling policies.
+ */
+typedef enum {
+  LF_SCHED_FAIR,      // Non real-time scheduling policy. Corresponds to SCHED_OTHER
+  LF_SCHED_TIMESLICE, // Real-time, time-slicing priority-based policty. Corresponds to SCHED_RR.
+  LF_SCHED_PRIORITY,  // Real-time, priority-only based scheduling. Corresponds to SCHED_FIFO.
+} lf_scheduling_policy_type_t;
+
+typedef struct {
+  lf_scheduling_policy_type_t policy; // The scheduling policy
+  int priority;                       // The priority, if applicable
+  interval_t time_slice;              // The time-slice allocated, if applicable.
+} lf_scheduling_policy_t;
+
+/**
+ * This pins a lf_thread to a specific CPU
+ *
+ * @param thread The thread
+ * @param cpu_number the CPU ID
+ * @return 0 on success, platform-specific error number otherwise.
+ */
+int lf_thread_set_cpu(lf_thread_t thread, int cpu_number);
+
+/**
+ * Sets the priority of a thread. Priority ranges from 0 to 99 where a higher
+ * number indicates higher priority. Setting the priority of a thread only
+ * makes sense if the thread is scheduled with LF_SCHED_TIMESLICE or LF_THREAD_PRIORITY
+ *
+ * @param thread The thread.
+ * @param priority The priority.
+ * @return int 0 on success, platform-specific error otherwise
+ */
+int lf_thread_set_priority(lf_thread_t thread, int priority);
+
+/**
+ * Sets the scheduling policy of a thread.
```

Comment:
One thing that would help me, at least, if this explains what it means to set a scheduling policy for a thread.  Doesn't a scheduling policy apply to a group of threads?  What if some threads have one scheduling policy and others have a different scheduling policy?  Maybe add a pointer to this page: https://man7.org/linux/man-pages/man7/sched.7.html. That page seems to explain it pretty well.  Also, we need to explain how the LF scheduling policies map to Linux scheduling policies, at least as an example.  E.g., `LF_SCHED_PRIORITY` to `SCHED_FIFO`.

Reply:
Yes thanks